`cargo-component` is out of use

`make tools` is available, but not functional:

```
asterinas $ make tools
error: current package believes it's in a workspace when it's not:
current:   /home/gh-zjp-CN/asterinas/kernel/libs/comp-sys/cargo-component/Cargo.toml
workspace: /home/gh-zjp-CN/asterinas/Cargo.toml

this may be fixable by adding `kernel/libs/comp-sys/cargo-component` to the `workspace.members` array of the manifest located at: /home/gh-zjp-CN/asterinas/Cargo.toml
Alternatively, to keep it out of the workspace, add the package to the `workspace.exclude` array, or add an empty `[workspace]` table to the package's manifest.
make: *** [Makefile:273: tools] Error 101
```

The cargo-component is broken: it relies on a 2023 toolchain while its local dependencies (like jhash) are in 2024 edition

```
libs/comp-sys/cargo-component $ cargo c
error: failed to load manifest for workspace member `/home/gh-zjp-CN/asterinas/kernel`

Caused by:
  failed to load manifest for dependency `aster-bigtcp`

Caused by:
  failed to load manifest for dependency `jhash`

Caused by:
  failed to parse manifest at `/home/gh-zjp-CN/asterinas/kernel/libs/jhash/Cargo.toml`

Caused by:
  failed to parse the `edition` key

Caused by:
  this version of Cargo is older than the `2024` edition, and only supports `2015`, `2018`, and `2021` editions.
```

cc 
* https://github.com/asterinas/asterinas/issues/348
* https://github.com/asterinas/asterinas/issues/58

So what's the plan for cargo-component?

--- a/Makefile
+++ b/Makefile
@@ -268,10 +268,6 @@ initramfs: check_vdso
 build: initramfs $(CARGO_OSDK)
 	@cd kernel && cargo osdk build $(CARGO_OSDK_BUILD_ARGS)
 
-.PHONY: tools
-tools:
-	@cd kernel/libs/comp-sys && cargo install --path cargo-component
-
 .PHONY: run
 run: initramfs $(CARGO_OSDK)
 	@cd kernel && cargo osdk run $(CARGO_OSDK_BUILD_ARGS)

